include/Helpers/String.hpp

```diff
@@ -153,16 +154,20 @@ namespace RC
 
     auto inline to_string(std::wstring& input) -> std::string
     {
-#pragma warning(disable: 4244)
-        return std::string{input.begin(), input.end()};
-#pragma warning(default: 4244)
+#pragma warning(disable: 4996)
+        std::string ascii_str(input.length(), '!');
+        std::wcstombs(&ascii_str[0], input.c_str(), input.length());
+        return ascii_str;
+#pragma warning(default: 4996)
     }
 
     auto inline to_string(std::wstring_view input) -> std::string
     {
-#pragma warning(disable: 4244)
-        return std::string{input.begin(), input.end()};
-#pragma warning(default: 4244)
+#pragma warning(disable: 4996)
+        std::string ascii_str(input.length(), '!');
+        std::wcstombs(&ascii_str[0], input.data(), input.length());
+        return ascii_str;
```

Comment:
is it against current code style to add some format of a null check here? 

if (ascii_str.empty()) ascii_str("string empty or null!");

Something like this?

Reply:
Since this is just a generic string converter, I think it would be better if the output is as close to the input as possible.
I think it would be more expected that if you convert a string, even if that string happened to be empty, you get an empty string back.

You're right about the null-check though, however, we should check the state of `input` and return early instead of checking the result of the conversion.